pkg/registration/register/interface.go

```diff
@@ -71,6 +71,9 @@ type RegisterDriver interface {
 	// InformerHandler returns informer of the related object. If no object needs to be watched, the func could
 	// return nil, nil.
 	InformerHandler(option any) (cache.SharedIndexInformer, factory.EventFilterFunc)
+
+	// AddClusterAnnotations adds cluster annotations for non-CSR drivers
+	AddClusterAnnotations(clusterAnnotations map[string]string, managedClusterArn string, managedClusterRoleSuffix string)
```

Comment:
I could understand the intension. But I wonder it is suitable for an interface since it seems quite specific to a certain implementation. I am thinking whether it makes more sense to have a method like

```
ClusterDecorator(cluster *v1.ManagedCluster) *v1.ManagedCluster
```

and call it in the creatingCluster controller. 

I can try to create an example PR to show how it looks like.

Reply:
https://github.com/open-cluster-management-io/ocm/pull/752

Reply:
@qiujian16 We are incorporating the changes from #752 into our changes. We'll let you know once we are ready for another around of reviews. Thanks for creating the example PR to demonstrate the changes.